### PR TITLE
[Teams] Add support for meeting participants added/removed events

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR adds support to _TeamsActivityHandler_ for meeting participants added/removed events.

## Specific Changes
- Added support for 2 new activity types - `application/vnd.microsoft.meetingParticipantJoin` and `application/vnd.microsoft.meetingParticipantLeave`.
- Added functions to register handlers for when teams meeting participants are added/removed.
- Added unit tests.

## Testing
These images show the new unit tests passing and the implementation tested with a sample.
image
image
image

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs [hunk 1]
```diff
@@ -857,6 +861,34 @@ protected virtual Task OnTeamsReadReceiptAsync(ReadReceiptInfo readReceiptInfo,
             return Task.CompletedTask;
         }
 
+        /// <summary>
+        /// Invoked when a Teams Participants Added event activity is received from the connector.
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Builder/Teams/TeamsActivityHandler.cs:865]: Update the comments on these 2 methods to match the new name of the event.

File: libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsJoinEventDetails.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsJoinEventDetails.cs:6]: System and Newtonsoft are not necessary.

File: libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsLeaveEventDetails.cs
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsLeaveEventDetails.cs:1]: I think these 2 classes MeetingParticipantsJoinEventDetails and MeetingParticipantsLeaveEventDetails can be removed. They are the same and don't add anything to the MeetingParticipantsEventDetails model.

File: libraries/Microsoft.Bot.Schema/Teams/TeamsMeetingMember.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Schema.Teams
+{
+    /// <summary>
+    /// Data about the joined meeting participants.
```
Comment (ceciliaavila) [inline libraries/Microsoft.Bot.Schema/Teams/TeamsMeetingMember.cs:9]: Is not just the joined participant. This class is used in both join and leave events. Review all the comments.

File: tests/Microsoft.Bot.Schema.Tests/Teams/MeetingParticipantsJoinEventDetailsTests.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+﻿// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Bot.Schema.Teams;
+using Xunit;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Schema.Tests/Teams/MeetingParticipantsJoinEventDetailsTests.cs:7]:
```suggestion
using System.Collections.Generic;
using Microsoft.Bot.Schema.Teams;
using Xunit;
```

File: tests/Microsoft.Bot.Schema.Tests/Teams/MeetingParticipantsJoinEventDetailsTests.cs
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Schema.Tests/Teams/MeetingParticipantsJoinEventDetailsTests.cs:1]: Probably these two test classes will be merged into one if we remove the other two classes MeetingParticipantsJoinEventDetails and MeetingParticipantsLeaveEventDetails.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsJoinEventDetails.cs (hunk 2 above)
Comment (JhontSouth) [reply libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsJoinEventDetails.cs:6]: Newtonsoft is being used for the JsonProperty tag

Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Schema/Teams/MeetingParticipantsJoinEventDetails.cs:6]: Which tag?
image

File: libraries/Microsoft.Bot.Schema/Teams/TeamsMeetingMember.cs (hunk 3 above)
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Schema/Teams/TeamsMeetingMember.cs:9]: This feedback wasn't applied.

**Review by ceciliaavila — APPROVED** [review]

File: tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs [hunk 5]
```diff
@@ -1270,6 +1270,82 @@ void CaptureSend(Activity[] arg)
...
+            // Arrange
+            var activity = new Activity
+            {
+                ChannelId = Channels.Msteams,
+                Type = ActivityTypes.Event,
+                Name = "application/vnd.microsoft.meetingParticipantJoin",
+                Value = JObject.Parse(@"{
+                    Members: [
+                        {User: {Id: 'id', Name: 'name'}, 
+                        Meeting: {Role: 'role', InMeeting: true}}
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.Tests/Teams/TeamsActivityHandlerTests.cs:1285]: I got confused about how this was formatted, i thought there were two items in the array, maybe we can improve that? Same in other places

**Review by sw-joelmut — APPROVED** [review]
Just a small detail, the rest lgtm

**Comment by ceciliaavila** [comment]
changes pushed to 6677

**Pull request closed without merging**